File: ClientApp/src/app/domain/product.ts
export interface IProduct{
    id: string;
    productName: string;
    price: number;
    origin: string;
    category: string;
    packaging: string;
    weight: number;
    unit: string;
	 stocklevel: number;
    imageUrl: string;
    productUrl: string;
   
    description: string;
	DetailDescription: string;
}
export class CementProduct{
 
}
export class Product{
    id: string;
    productName: string;
    price: number;
    origin: string;
    category: string;
    packaging: string;
    weight: number;
    unit: string;
    imageUrl: string;
    productUrl: string;
    stocklevel: number;
    description: string;
}

export class VProduct{
    pro: string;

}
export class ProductSale{
    id: string;
    productName: string;
    price: number;
    origin: string;
    packaging: string;
    weight: number;
    unit: string;
    imgUrl: string;
    productUrl: string;
}
export class IMomoProvider{

}
export class MomoProvider{
    
}
export class RelatedProducts{
    productName: string;
    imageUrl: string;
    productUrl: string

}
export interface IRelatedProducts{
    productName: string;
    imageUrl: string;
    productUrl: string;

}

File: Entities/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Aggregate_Direct_ShiaHill.Entities
{
    public class Product
    {
        [Key]
        public Guid ID { get; set; }
        [ForeignKey("Supplier")]
        [Required]
        public string SupplierID { get; set; }
       // [ForeignKey("ProductCategory")]
        //[Required]
        //public int ProductCategoryID { get; set; }
        [Required]
        public string ProductName { get; set; }

        [Required]
        [StringLength(4)]
        public string EmployeeInitials { get; set; }

        [Required]
        public decimal Price { get; set; }
        [Required]
        public string Origin { get; set; }
        [Required]
        public string Category { get; set; }
        [Required]
        public string Packaging { get; set; }
        public double WeightPerUnit { get; set; }
        [Required]
        [StringLength(maximumLength:2)]
        public string Unit { get; set; }
        [Required]
        public int StockLevel { get; set; }
        [Required]
        [DataType(DataType.DateTime)]
        [DisplayFormat(DataFormatString = "{0:dd/MM/yyyy}")]
        public DateTime EntryDate { get; set; }
        [Required]
        public string Description { get; set; }
        [Required]
        public string DetailDescription { get; set; }
        [Required]
        public string ImageUrl { get; set; }
        [Required]
        public string ProductUrl { get; set; }

        public Supplier Supplier { get; set; }
    }
}
